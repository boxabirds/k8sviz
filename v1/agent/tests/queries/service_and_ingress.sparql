# Query: Create a Service and Ingress for external access
# This query retrieves information for exposing an application

PREFIX k8s: <http://kubernetes.io/ontology#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT ?resource ?property ?value ?comment
WHERE {
  # Get Service information
  {
    k8s:Service a owl:Class ;
                rdfs:comment ?serviceComment .
    
    ?serviceType a k8s:ServiceType ;
                 rdfs:label ?serviceTypeLabel ;
                 rdfs:comment ?serviceTypeComment .
  }
  
  # Get Service properties
  UNION
  {
    ?property rdfs:domain k8s:Service ;
              rdfs:label ?propertyLabel ;
              rdfs:comment ?propertyComment .
  }
  
  # Get selector information
  UNION
  {
    k8s:Selector rdfs:comment ?selectorComment .
    k8s:selectsPod rdfs:comment ?selectsPodComment .
  }
  
  # Get networking best practices
  UNION
  {
    ?practice a k8s:BestPractice ;
              rdfs:comment ?practiceComment .
    FILTER(CONTAINS(LCASE(STR(?practiceComment)), "service") || 
           CONTAINS(LCASE(STR(?practiceComment)), "network"))
  }
}