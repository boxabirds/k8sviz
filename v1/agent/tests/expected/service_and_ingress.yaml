---
apiVersion: v1
kind: Service
metadata:
  name: web-app-service
spec:
  type: ClusterIP
  selector:
    app: web-app
  ports:
  - port: 80
    targetPort: 80
    protocol: TCP
    name: http
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: web-app-ingress
  annotations:
    nginx.ingress.kubernetes.io/rewrite-target: /
spec:
  ingressClassName: nginx
  rules:
  - host: web-app.example.com
    http:
      paths:
      - path: /
        pathType: Prefix
        backend:
          service:
            name: web-app-service
            port:
              number: 80
  tls:
  - hosts:
    - web-app.example.com
    secretName: web-app-tls